build.gradle.kts

```
@@ -10,3 +12,24 @@ plugins {
     alias(libs.plugins.ksp) apply false
     alias(libs.plugins.room) apply false
 }
+
+val versionProperties = loadPropertiesFromFile(file("version.properties"))
+
+project.apply {
+    extra.apply {
+        set("versionName", versionProperties.getProperty("versionName", "0.1"))
+        set("versionCode", versionProperties.getProperty("versionCode", "1").toInt())
```

❓ Should we not hardcode those values?
```suggestion
        set("versionName", versionProperties.getProperty("versionName", null))
        set("versionCode", versionProperties.getProperty("versionCode", null).toInt())
```

---

Agreed, no point of having a default -- `version.properties` should always be there and properly configured, otherwise things should fail.